lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -527,7 +531,7 @@ def name(self):
 
     @property
     def canonical_package_id(self):
-        if not self.is_virtual:
+        if not self.is_virtual or self.is_ansible:
```

Comment:
`if self.is_ansible` should probably generate a different string too (not '<virtual namespace ...>' but maybe '<ansible-core runtime>'. Otherwise, resolvelib would treat virtual namespaces as equal to runtimes.

```suggestion
        if self.is_ansible:
            return '<ansible-core runtime>'

        if not self.is_virtual:
```

Reply:
This expression is handling non-virtual candidates and runtimes the same way. Resolvelib won't treat non-virtual as equal to runtimes because 1) find_matches handles this type specifically 2) it cannot match an actual fqcn since there is no '.' in it.

In the last commit, I changed the canonical_package_id and displayed name from "Ansible" to "ansible-core".